### PR TITLE
Adding timeshift units

### PR DESCRIPTION
**Pull request by Rasinhas**
Added a dropdown element to enable changing the period unit on timeshift. 

Unit options are: seconds, minutes, hours, days, weeks and months.

timeshiftdd

### CODE REVIEW

File: src/query_ctrl.ts [hunk 1]
```diff
@@ -38,6 +38,8 @@ export class MetaQueriesQueryCtrl extends QueryCtrl {
 
     defaultPeriods = 7;
 
+    defaultUnit = "days";
```
Comment (Gauravshah) [inline src/query_ctrl.ts]: 👍

File: src/query_ctrl.ts [hunk 2]
```diff
@@ -52,6 +54,9 @@ export class MetaQueriesQueryCtrl extends QueryCtrl {
       if (!this.target.periods) {
         this.clearPeriods();
       }
+      if (!this.target.units) {
+      this.target.units = this.defaultUnit;
```
Comment (Gauravshah) [inline src/query_ctrl.ts]: indentation

File: src/query_ctrl.ts [hunk 3]
```diff
@@ -52,6 +54,9 @@ export class MetaQueriesQueryCtrl extends QueryCtrl {
       if (!this.target.periods) {
         this.clearPeriods();
       }
+      if (!this.target.units) {
```
Comment (Gauravshah) [inline src/query_ctrl.ts]: calling this `timeshiftUnit` might be better. I do understand existing code doesn't do it for periods but would love to scope it to type unless its generic. keeping in mind about stddev etc

File: src/query_ctrl.ts (hunk 1 above)
Comment (Gauravshah) [reply src/query_ctrl.ts]: default helps not breaking existing ones

**Comment by Gauravshah** [comment]
@Rasinhas thanks for the PR

File: src/query_ctrl.ts (hunk 2 above)
Comment (Rasinhas) [reply src/query_ctrl.ts]: fixed

File: src/query_ctrl.ts (hunk 3 above)
Comment (Rasinhas) [reply src/query_ctrl.ts]: done

**Comment by Gauravshah** [comment]
fixes #22

File: src/datasource.js [hunk 4]
```diff
@@ -248,7 +249,7 @@ function (angular, _, dateMath, moment) {
               data.forEach(function (datum) {
                   if(datum.target===metric){
                     datum.datapoints.forEach(function (datapoint) {
-                        datapoint[1] = dateToMoment(new Date(datapoint[1]),false).subtract(periodsToShift,'days').toDate().getTime();
+                        datapoint[1] = dateToMoment(new Date(datapoint[1]),false).subtract(periodsToShift,unit).toDate().getTime();
```
Comment (Gauravshah) [inline src/datasource.js]: I missed looking at this, can you fix variables and raise another PR ?

File: dist/datasource.js [hunk 5]
```diff
@@ -221,9 +221,10 @@ function (angular, _, dateMath, moment) {
         var periodsToShift = target.periods;
         var query = target.query;
         var metric = target.metric;
+        var unit = target.unit;
```
Comment (Gauravshah) [inline dist/datasource.js]: same here

File: dist/query_ctrl.ts [hunk 6]
```diff
@@ -38,6 +38,8 @@ export class MetaQueriesQueryCtrl extends QueryCtrl {
 
     defaultPeriods = 7;
 
+    defaultUnit = "days";
```
Comment (Gauravshah) [inline dist/query_ctrl.ts]: defaultTimeShiftUnit

**Comment by Gauravshah** [comment]
@Rasinhas I have reverted the code changes on master, please open a new PR with fixes

**Pull request merged**